deps/coro-channel.lua

```diff
@@ -120,10 +120,18 @@ end
 
 local function makeWrite(socket, closer)
 
+  local hasYielded, hasReturned
+  local success, err
```

Comment:
These variables hold state across writes and are not cleared before each use. This is a bug, after the first instantaneous write `hasReturned` will always be true.

Reply:
Thanks for catching this one! I honestly was worried about it since the last time I tested a fix like this it completely killed my performance testing.  This was likely the culprit!

Should I just introduce an upvalue `resetStates` that gets called before write/close calls?

Reply:
Should be fixed. On `wait` calls we now reset the states, `wait` is called before any write/shutdown call, so it ensures the states always reset.

Expanded the tests to catch this mistake, the "mixed instant/delayed writes" test will now catch it.